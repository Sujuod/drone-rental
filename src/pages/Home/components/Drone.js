import React from 'react'
import { isDroneRented } from 'utilities/droneStorage';

const rentBtnStyle = {
    fontWeight: '700',
    background: '#690513',
    color: 'white'
}

const Drone = ({ info, onRentClick }) => {
    const rent = () => {
        onRentClick(info);
    }

    return (
        <div className=" col-md-4 col-sm-6" >
            <div className="card mt-3" style={cardStyle}>
                <div className="text-center">
                    <img className="card-img-top w-50" src={require(`assets/images/${info.manufacturer}.jpg`)} alt="Card name" />
                </div>
                <div className="card-body text-center">
                    <div className="card-text">
                        <h5 className="card-title">{info.model}</h5>
                        <ul className="list-group">
                            <li className="list-group-item">Manufacturer : {info.manufacturer}</li>
                            <li className="list-group-item">Max Flight Time : {info.maxFlightTime}</li>
                            <button type="btn" className="list-group-item list-group-item-action" onClick={rent} style={rentBtnStyle} >Rent</button>
                        </ul>
                    </div>
                </div>
                {isDroneRented(info.model) && <div style={ifRented}>RENTED</div>}
            </div>
        </div>
    )
}
export default Drone;

const cardStyle = {
    position: 'relative'
}

const ifRented = {
    position: 'absolute',
    width: '100%',
    height: '100%',
    top: '0',
    left: '0',
    background: 'black',
    opacity: '0.7',
    color: 'white',
    fontSize: '60px',
    justifyContent: 'center',
    display: 'flex',
    alignItems: 'center'
}