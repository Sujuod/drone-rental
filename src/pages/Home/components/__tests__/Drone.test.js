import Drone from '../Drone';

function setup() {
  const info = {
        "manufacturer": "Parrot",
        "model": "Anafi",
        "maxFlightTime": "1min",
        "charge": "0%"
    }

  const component = shallow(<Drone info={info}/>);

  return component;
}

describe('Drone component', () => {
  const component = setup();

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });
});