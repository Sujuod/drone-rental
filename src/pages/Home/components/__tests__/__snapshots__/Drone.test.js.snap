// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardItem component should render correctly 1`] = `
<div
  className=" col-md-4 col-sm-6"
>
  <div
    className="card mt-3"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="text-center"
    >
      <img
        alt="Card name"
        className="card-img-top w-50"
        src="Parrot.jpg"
      />
    </div>
    <div
      className="card-body text-center"
    >
      <div
        className="card-text"
      >
        <h5
          className="card-title"
        >
          Anafi
        </h5>
        <ul
          className="list-group"
        >
          <li
            className="list-group-item"
          >
            Manufacturer : 
            Parrot
          </li>
          <li
            className="list-group-item"
          >
            Max Flight Time : 
            1min
          </li>
          <button
            className="list-group-item list-group-item-action"
            onClick={[Function]}
            style={
              Object {
                "background": "#690513",
                "color": "white",
                "fontWeight": "700",
              }
            }
            type="btn"
          >
            Rent
          </button>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drone component should render correctly 1`] = `
<div
  className=" col-md-4 col-sm-6"
>
  <div
    className="card mt-3"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="text-center"
    >
      <img
        alt="Card name"
        className="card-img-top w-50"
        src="Parrot.jpg"
      />
    </div>
    <div
      className="card-body text-center"
    >
      <div
        className="card-text"
      >
        <h5
          className="card-title"
        >
          Anafi
        </h5>
        <ul
          className="list-group"
        >
          <li
            className="list-group-item"
          >
            Manufacturer : 
            Parrot
          </li>
          <li
            className="list-group-item"
          >
            Max Flight Time : 
            1min
          </li>
          <button
            className="list-group-item list-group-item-action"
            onClick={[Function]}
            style={
              Object {
                "background": "#690513",
                "color": "white",
                "fontWeight": "700",
              }
            }
            type="btn"
          >
            Rent
          </button>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
